src/Symfony/Component/DependencyInjection/Loader/IniFileLoader.php

```diff
@@ -52,7 +52,15 @@ public function load($resource, $type = null)
      */
     public function supports($resource, $type = null)
     {
-        return is_string($resource) && 'ini' === pathinfo($resource, PATHINFO_EXTENSION);
+        if (!is_string($resource)) {
+            return false;
+        }
+
+        if (null === $type && 'ini' === pathinfo($resource, PATHINFO_EXTENSION)) {
+            return true;
+        }
+
+        return null !== $type && 'ini' === $type;
```

Comment:
the type will have priority if explicitly provided with current implementation, as the above condition has a `null === $type` check.

The difference with your snippet is that the loader won't return `true` in case the type was explicitly provided, but not `ini` and the extension is `ini` (another loader should try)